cmd/oras/internal/option/remote.go

```diff
@@ -305,9 +313,12 @@ func (opts *Remote) NewRepository(reference string, common Common, logger logrus
 
 // isPlainHttp returns the plain http flag for a given registry.
 func (opts *Remote) isPlainHttp(registry string) bool {
+	if plainHTTP := opts.getPlainHTTP(); plainHTTP != nil {
+		return *plainHTTP
+	}
 	host, _, _ := net.SplitHostPort(registry)
 	if host == "localhost" || registry == "localhost" {
 		return true
 	}
-	return opts.PlainHTTP
+	return false
```

Comment:
This code can be simplified as `return host == "localhost" || registry == "localhost"`